### PR TITLE
Replace hardcoded z-level numbers with a trait system

### PR DESCRIPTION
**Pull request by SpaceManiac**
:cl:
refactor: Information about the properties of z-levels is now less hardcoded than before.
/:cl:

Done in this PR:

* Add trait system from #32037 and prepare file layout for the future
* Update defines introduced in #33829 to use traits
* Define traits for the maps in `basemap.dm`, the autogenerated deep space, and away missions
* Use traits to handle bombcap increase on Lavaland and decrease on Reebe
* Use traits to know which z-levels to spawn lava or space ruins on
* Replace `GLOB.station_z_levels` entirely by traits
* Eliminate some but not all hardcoded z-level checks
* Add a "Debug Z-Levels" verb which shows all z-level info and a map of deep space

I specifically tested space transitions to make sure they still work. Generally speaking, no player-facing behavior should have changed. There's a lot more to go on the z-level stuff but this is another midpoint where PR-worthy work has been done and everything is in a functional state.

### CODE REVIEW

File: code/__HELPERS/level_traits.dm [hunk 1]
```diff
@@ -1,17 +1,17 @@
 // Helpers for checking whether a z-level conforms to a specific requirement
 
 // Basic levels
-#define is_centcom_level(z) ((z) == ZLEVEL_CENTCOM)
+#define is_centcom_level(z) SSmapping.level_trait(z, ZTRAIT_CENTCOM)
```
Comment (MoreRobustThanYou) [inline code/__HELPERS/level_traits.dm]: you forgot a ```(``` before the SSmapping.

**Comment by Cyberboss** [comment]
Thanks for all this work! This'll make the port so much less painful

**Comment by Crazylemon64** [comment]
woaaaah
🎉 it's happening 🎉

**Comment by optimumtact** [comment]
hey slowpoke

**Pull request merged**